### ISSUE TITLE
Update GSI fix submodule to bring in GFS v16.3.9 changes

### ISSUE THREAD

**Issue by RussTreadon-NOAA** [NOAA-EMC/GSI#620]
GFS v16.3.9 was implemented 20230911 12Z.   GFS issue [#1813](https://github.com/NOAA-EMC/global-workflow/issues/1813) provides details.

Three GSI related changes were implemented
1. turn off ingest of NOAA-21 radiances into the GSI,
2. assimilate PlanetIQ GPSRO and NOAA-20 OMPS (Ozone Mapping
and Profiler Suite) Ozone retrievals
3. adjust OMPS observation errors

Change 1 modified `exglobal_atmos_analysis.sh`.   This script no longer resides in the GSI repository.  g-w maintains this script in their repository.  g-w issue [#1857](https://github.com/NOAA-EMC/global-workflow/issues/1857) has been opened to add gfs.v16.3.9 changes to this script.

Changes 2 and 3 are fix file changes.  `global_convinfo.txt`, and `global_ozinfo.txt` are modified.  ASCII GSI fix files are maintained in the GSI-fix repository.   GSI-fix issue [#9](https://github.com/NOAA-EMC/GSI-fix/issues/9) documents the updating of these two fix files.

This issue is opened to update the `fix` submodule hash to point at GSI-fix commit [5722cd4](https://github.com/NOAA-EMC/GSI-fix/commit/5722cd4d2519222137c5b356bdbc01bb34c5f1f4).  This hash includes the GFS v16.3.9 fix file updates.

**Comment by RussTreadon-NOAA** [comment]
Create [RussTreadon-NOAA/GSI:feature/updatefix](https://github.com/RussTreadon-NOAA/GSI/tree/feature/updatefix) in which to update fix submodule hash.

**Comment by RussTreadon-NOAA** [comment]
**Orion ctests**
Install `feature/updatefix` on Orion.  Run ctests with following results
```
Orion-login-3:/work2/noaa/da/rtreadon/git/gsi/updatefix/build$ ctest -j 9
Test project /work2/noaa/da/rtreadon/git/gsi/updatefix/build
    Start 1: global_3dvar
    Start 2: global_4dvar
    Start 3: global_4denvar
    Start 4: hwrf_nmm_d2
    Start 5: hwrf_nmm_d3
    Start 6: rtma
    Start 7: rrfs_3denvar_glbens
    Start 8: netcdf_fv3_regional
    Start 9: global_enkf
1/9 Test #8: netcdf_fv3_regional ..............***Failed  483.12 sec
2/9 Test #5: hwrf_nmm_d3 ......................   Passed  495.96 sec
3/9 Test #9: global_enkf ......................***Failed  551.96 sec
4/9 Test #7: rrfs_3denvar_glbens ..............***Failed  606.53 sec
5/9 Test #4: hwrf_nmm_d2 ......................   Passed  608.96 sec
6/9 Test #2: global_4dvar .....................   Passed  2102.39 sec
7/9 Test #3: global_4denvar ...................***Failed  2102.67 sec
8/9 Test #6: rtma .............................   Passed  2230.80 sec
9/9 Test #1: global_3dvar .....................   Passed  2584.31 sec

56% tests passed, 4 tests failed out of 9

Total Test time (real) = 2584.32 sec

The following tests FAILED:
          3 - global_4denvar (Failed)
          7 - rrfs_3denvar_glbens (Failed)
          8 - netcdf_fv3_regional (Failed)
          9 - global_enkf (Failed)
```
The **global_4denvar** failure is due to
```
The runtime for global_4denvar_loproc_updat is 445.255430 seconds.  This has exceeded maximum allowable threshold time of 443.334139 seconds,
resulting in Failure time-thresh of the regression test.
```
The `gsi.x` wall times do not reveal any anomalous behavior
```
global_4denvar_hiproc_contrl/stdout:The total amount of wall time                        = 314.898718
global_4denvar_hiproc_updat/stdout:The total amount of wall time                        = 301.392255
global_4denvar_loproc_contrl/stdout:The total amount of wall time                        = 403.031036
global_4denvar_loproc_updat/stdout:The total amount of wall time                        = 445.255430
```
The _updat_ is faster at the high task count and a slower at the low task count.   These timing results could change with additional runs of the global_4denvar test.  **This is not a fatal fail.**

The **rrfs_3denvar_glbens** failure is due to
```
The case has Failed the scalability test.
The slope for the update (37.419338 seconds per node) is less than that for the control (39.897894 seconds per node).
```
A check of the `gsi.x` wall time indicates comparable wall times with _updat_ running a bit faster
```
rrfs_3denvar_glbens_hiproc_contrl/stdout:The total amount of wall time                        = 100.611999
rrfs_3denvar_glbens_hiproc_updat/stdout:The total amount of wall time                        = 98.145926
rrfs_3denvar_glbens_loproc_contrl/stdout:The total amount of wall time                        = 140.509893
rrfs_3denvar_glbens_loproc_updat/stdout:The total amount of wall time                        = 128.081397
```
**This is not a fatal fail.**

The **netcdf_fv3_regional** failure is due to 
```
The memory for netcdf_fv3_regional_loproc_updat is 295168 KBs.  This has exceeded maximum allowable memory of 272782 KBs,
resulting in Failure memthresh of the regression test.
```
A check of the task 0 resident set size does not reveal anomalous behavior
```
netcdf_fv3_regional_hiproc_contrl/stdout:The maximum resident set size (KB)                   = 238776
netcdf_fv3_regional_hiproc_updat/stdout:The maximum resident set size (KB)                   = 293764
netcdf_fv3_regional_loproc_contrl/stdout:The maximum resident set size (KB)                   = 247984
netcdf_fv3_regional_loproc_updat/stdout:The maximum resident set size (KB)                   = 295168
```
**This is not a fatal fail.**

The global_enkf failure is due to
```
The runtime for global_enkf_hiproc_updat is 79.030027 seconds.  This has exceeded maximum allowable threshold time of 76.400589 seconds,
resulting in Failure timethresh2 of the regression test.
```
A check of the `enkf.x` wall times does not indicate any anomalous behavior.
```
global_enkf_hiproc_contrl/stdout:The total amount of wall time                        = 69.455081
global_enkf_hiproc_updat/stdout:The total amount of wall time                        = 79.030027
global_enkf_loproc_contrl/stdout:The total amount of wall time                        = 95.662371
global_enkf_loproc_updat/stdout:The total amount of wall time                        = 99.798867
```
**This is not a fatal fail.**

**Summary:**  For all tests _updat_ and _contrl_ generate identical analyses.   This result is expected for global_enkf, hwrf_nmm_d2, hwrf_nmm_d3, netcdf_fv3_regional, rrfs_3denvar_glbens, and rtma.  None of these tests assimilate ompsnp or PlanetIQ gpsro.

Differences are expected for global_3dvar, global_4denvar, and global_4dvar.  None of these tests assimilate PlanetIQ gpsro but each test does assimilate OMPSNP data.   The _updat_ and _contrl_ analyses do not differ because both sets of runs use the same fix files, namely the _updat_ fix files.

Modify `global_3dvar.sh` to use different fix files for _updat_ and _contrl_.  Rerun the global_3dvar.   Confirm that the _updat_ and _contrl_ jobs use different `convinfo` and `ozinfo`.  The `convinfo` difference has no impact on analysis results as explained above.   The `ozinfo` difference alters analysis results due to different observation errors being used for assimilated ompsnp data.   This is expected behavior.

**Comment by RussTreadon-NOAA** [comment]
While executing `ush/build.sh` I was reminded that the build copies GSI binary fix files from the EID staged directory into the local working copy of `fix`.   For example, on WCOSS2 we see the following as `build.sh` executes
```
-- Found w3emc: /apps/ops/prod/libs/intel/19.1.3.304/w3emc/2.9.1/lib64/libw3emc_4.a (found version "2.9.1")
-- Copying GSI binary fix files from /lfs/h2/emc/global/noscrub/emc.global/FIX/fix/gsi/20230601
-- RT: Control GSIexec Found: /lfs/h2/emc/da/noscrub/russ.treadon/git/gsi/develop/install/bin/gsi.x
```
Note that GSI binary fix files are being copied from a directory named `2023061`.   g-w PR [#1858](https://github.com/NOAA-EMC/global-workflow/pull/1858) updated the EID GSI fix directory to `20230911`

GSI modulefiles define `GSI_BINARY_SOURCE_DIR`.   This variable currently points at the `20230601` directory.  Since this issue updates the GSI-fix hash, we should also update the `GSI_BINARY_SOURCE_DIR` to `20230911`.

This change has been made to `$MACHINE` specific `modulefiles/gsi_$MACHINE.lua`.   Done at 1176751f.
